src/components/mapPageComponents/ducks/test/thunks.test.ts

```diff
@@ -0,0 +1,180 @@
+      const mockGetSiteGeoData = jest.fn();
+      const mockBlockDataResponse: BlockGeoData = {
+        type: 'FeatureCollection',
+        name: 'blocks',
+        features: [
+          {
+            type: 'Feature',
+            properties: {
+              blockId: 1714,
+              lat: 42.3488784985,
+              lng: -71.0293810011,
+            },
+            geometry: {
+              type: 'MultiPolygon',
+              coordinates: [
+                [
+                  [
+                    [-71.02859399925286, 42.34889700150278],
+                    [-71.02912299801895, 42.34837799993562],
+                  ],
+                ],
+              ],
+            },
+          },
+        ],
+      };
+      const mockNeighborhoodDataResponse: NeighborhoodGeoData = {
+        type: 'FeatureCollection',
+        name: 'neighborhoods',
+        features: [
+          {
+            type: 'Feature',
+            properties: {
+              neighborhoodId: 15,
+              name: 'Roslindale',
+              completionPerc: 0,
+              lat: 42.2803,
+              lng: -71.1266,
+            },
+            geometry: {
+              type: 'MultiPolygon',
+              coordinates: [
+                [
+                  [
+                    [-71.12592717485386, 42.272013107957406],
+                    [-71.12610933458738, 42.2716219294518],
+                  ],
+                ],
+              ],
+            },
+          },
+        ],
+      };
+      const mockSiteDataResponse: SiteGeoData = {
+        type: 'FeatureCollection',
+        name: 'sites',
+        features: [
+          {
+            type: 'Feature',
+            properties: {
+              id: 213,
+              treePresent: true,
+              diameter: 29.8,
+              species: 'Willow',
+              updatedAt: 'December 17, 1995 03:24:00',
+              updatedBy: 'username',
+              address: '123 Street',
+              lat: 42.3488784985,
+              lng: -71.0293810011,
+            },
+            geometry: {
+              type: 'Point',
+              coordinates: [[[[-71.02859399925286, 42.34889700150278]]]],
+            },
+          },
+        ],
+      };
+
+      mockGetBlockGeoData.mockResolvedValue(mockBlockDataResponse);
+      mockGetNeighborhoodGeoData.mockResolvedValue(
+        mockNeighborhoodDataResponse,
+      );
+      mockGetSiteGeoData.mockResolvedValue(mockSiteDataResponse);
+      const mockExtraArgs: ThunkExtraArgs = {
+        authClient,
+        protectedApiClient,
+        apiClient: {
+          ...apiClient,
+          getBlockGeoData: mockGetBlockGeoData,
+          getNeighborhoodGeoData: mockGetNeighborhoodGeoData,
+          getSiteGeoData: mockGetSiteGeoData,
+        },
+      };
+
+      await getMapGeoData()(mockDispatch, getState, mockExtraArgs);
+
+      expect(mockDispatch).toHaveBeenCalledTimes(6);
+      expect(mockDispatch).toHaveBeenNthCalledWith(
+        4,
```

Comment:
❓ Not super familiar with these kinds of tests. What does this number signify?

Reply:
These probably explain it better than I could:
- [toHaveBeenCalledTimes](https://jestjs.io/docs/expect#tohavebeencalledtimesnumber)
- [toHaveBeenNthCalledWith](https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-)

but basically I'm checking that the mock get map geo data dispatch was called 6 times, and on the 4th call it was called with `blockGeoData.loaded(mockBlockDataResponse)` as the argument (the first three calls are the different `.loading` states).